libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -370,7 +583,28 @@ MQTTStatus_t MQTT_SerializePublishHeader( const MQTTPublishInfo_t * const pPubli
                                           const MQTTFixedBuffer_t * const pBuffer,
                                           size_t * const pHeaderSize )
 {
-    return MQTTSuccess;
+    MQTTStatus_t status = MQTTSuccess;
```

Comment:
As above, initializing to `MQTTBadParameter` allows the code to be shorter and the happy path to be in the `if`s

Reply:
replied in the above comment.